src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -245,8 +245,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ItemGroup>
 
       <!-- Copy all the assemblies -->
-      <ResolvedFileToPublish Include="@(ResolvedAssembliesToPublish)">
-      </ResolvedFileToPublish>
+        <ResolvedFileToPublish Include="@(ResolvedAssembliesToPublish)" Condition="'$(SkipRemovingSystemFiles)' == 'true' Or !$([System.Text.RegularExpressions.Regex]::IsMatch(%(Filename), '^System.'))" />
```

Comment:
Consider using `String.StartsWith` with a culture-invariant comparison. The current `Regex.IsMatch` test does not do what is intended (the `.` character is a regular expression wildcard that will match anything). 

Try the following code in the C# Interactive window; it returns `true`: 

```csharp
> using System.Text.RegularExpressions;
> Console.WriteLine(Regex.IsMatch("SystemX", "^System."));
True
```
https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/